src/SFML/Audio/Sound.cpp

```diff
@@ -36,21 +36,18 @@
 namespace sf
 {
 ////////////////////////////////////////////////////////////
-Sound::Sound() = default;
-
-
-////////////////////////////////////////////////////////////
-Sound::Sound(const SoundBuffer& buffer)
+Sound::Sound(const SoundBuffer& buffer) : m_buffer(&buffer)
 {
-    setBuffer(buffer);
+    m_buffer->attachSound(this);
+    alCheck(alSourcei(m_source, AL_BUFFER, static_cast<ALint>(m_buffer->m_buffer)));
```

Comment:
Not really a deal breaker, but we've duplicated these two lines in three places now.

Reply:
I agree this duplication it is not extremely elegant. The motivation here is that only constructors are responsible for setting up the invariant (i.e. sound it attached to a sound buffer), and methods (especially setBuffer) can be simpler by assuming the invariant is true.